console/client/src/features/modules/ModuleDetails.tsx

```diff
@@ -1,15 +1,30 @@
-import { useContext } from 'react'
-import { useLocation, useNavigate, useSearchParams } from 'react-router-dom'
+import React from 'react'
 import { SelectedModuleContext } from '../../providers/selected-module-provider'
-import { TabSearchParams, TabType, TabsContext } from '../../providers/tabs-provider'
-import { textColor } from '../../utils/style.utils'
+import { TabType, TabsContext, TabSearchParams } from '../../providers/tabs-provider'
+import { textColor, urlSearchParamsToObject } from '../../utils'
+import { useSearchParams } from 'react-router-dom'
+import { modulesContext } from '../../providers/modules-provider'
 
 export function ModuleDetails() {
-  const { selectedModule } = useContext(SelectedModuleContext)
-  const { tabs, setTabs, setActiveTab } = useContext(TabsContext)
-  const navigate = useNavigate()
-  const location = useLocation()
-  const [ searchParams ] = useSearchParams()
+  const modules = React.useContext(modulesContext)
+  const { selectedModule, setSelectedModule } = React.useContext(SelectedModuleContext)
+  const { tabs, setTabs, setActiveTab } = React.useContext(TabsContext)
+  const [ searchParams, setSearchParams ] = useSearchParams()
+  const moduleId = searchParams.get('module')
+  //P1 when mounting with a valid module in query params set selected module
```

Comment:
What are the `P1` `P2` type comments. Are they param 1,2? Maybe we can add more context here to help?

Reply:
Honestly I just like the letter P no reason other than maybe Priority. I use it to make notes about exception and edge case handling. I use P to differentiate from a basic comment notes describing functionality. Guess I could use Condition | Priority | Case | Exception specifically.